### PR TITLE
[ML Folder] Edit move error fixes

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

- Fixed EditFolderDialog form layout
- Fixed error message appearance for moving folder with EditFolderDialog
- Added danger border to React Select component on error
- Added test coverage for moving folder error

### CODE REVIEW

File: packages/core/helper-plugin/lib/src/components/ReactSelect/ReactSelect.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import getSelectStyles from './utils/getSelectStyles';
 
 const ReactSelect = ({ components, styles, ...props }) => {
   const theme = useTheme();
-  const customStyles = getSelectStyles(theme);
+  const customStyles = getSelectStyles(theme, props);
```
Comment (ronronscelestes) [inline packages/core/helper-plugin/lib/src/components/ReactSelect/ReactSelect.js:14]: @gu-stav didn't find another way to access error and applying style but not sure it's the best, do you have another solution in mind?

File: packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const getSelectStyles = theme => {
+const getSelectStyles = (theme, { 'aria-errormessage': error }) => {
```
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js:1]: Mh, I'm not a fan of passing in the error message through aria. Should we pass it in via an `error` prop and do the `aria-` handling inside of `ReactSelect` instead?

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js [hunk 3]
```diff
@@ -42,13 +41,12 @@ export const EditFolderDialog = ({ onClose, folder, parentFolderId }) => {
...
   const { canCreate, isLoading: isLoadingPermissions, canUpdate } = useMediaLibraryPermissions();
-  const submitButtonRef = useRef(null);
   const [showConfirmDialog, setShowConfirmDialog] = useState(false);
   const { formatMessage, formatDate } = useIntl();
-  const { editFolder, isLoading: isEditFolderLoading } = useEditFolder();
+  const { editFolder } = useEditFolder();
   const { remove } = useBulkRemove();
   const toggleNotification = useNotification();
-  const isLoading = isLoadingPermissions || folderStructureIsLoading || isEditFolderLoading;
+  const isLoading = isLoadingPermissions || folderStructureIsLoading;
```
Comment (gu-stav) [inline packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:49]: Why did you remove the `isEditFolderLoading`? The loading state needs to be displayed, if a request is happening I think.

Comment (ronronscelestes) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:49]: the formik error message wasn't displayed because when you submitted isEditFolderLoading was removing all inputs to show the loading state. 
that's the only solution I found to make the error state/message working, and it doesn't seem to break anything - don't hesitate to test and tell me if it's breaking anything! we can add a loading state to the save button though, we do it in many forms in Strapi, WDYT?

Comment (gu-stav) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:49]: Mh, maybe we need split the logic then, because I find it still helpful to see the loading state on e.g. the submit button. Otherwise the user doesn't have an indication of what is going on.

Could you add it back there, if you agree?

File: packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js (hunk 2 above)
Comment (ronronscelestes) [reply packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js:1]: actually the `aria` works / is handled by `ReactSelect` I think (see below), so I don't think we need to add anymore more an `aria` side!

image

for the error style, sure I can pass an error prop to getSelectStyles instead!

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 3 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:49]: sorry not sure I understand, do you mean adding a loading state to the submit button?

File: packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js (hunk 2 above)
Comment (gu-stav) [reply packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js:1]: What I meant was slightly different: I thought it might be helpful, if we pass an `error` prop to `ReactSelect` and render the `aria-errormessage` and `aria-invalid` props there, if an error was passed. This would make the component more compliant with the others and avoid some duplication in our forms ...

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 3 above)
Comment (gu-stav) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:49]: Yes. Not showing the loader seems fine to me, but I'd like to have the state still on the submit button at least.

File: packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js (hunk 2 above)
Comment (ronronscelestes) [reply packages/core/helper-plugin/lib/src/components/ReactSelect/utils/getSelectStyles.js:1]: sure will do!

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 3 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:49]: fixed [here](https://github.com/strapi/strapi/pull/13494/commits/a23a176e760cbfd309f91dd3d6a7f9bbc3ecdd3b)

File: packages/core/helper-plugin/lib/src/components/ReactSelect/ReactSelect.js [hunk 4]
```diff
@@ -9,14 +9,16 @@ import IndicatorSeparator from './components/IndicatorSeparator';
...
+const ReactSelect = ({ components, styles, error, ...props }) => {
   const theme = useTheme();
-  const customStyles = getSelectStyles(theme);
+  const customStyles = getSelectStyles(theme, error);
 
   return (
     <Select
       {...props}
       components={{ ClearIndicator, DropdownIndicator, IndicatorSeparator, ...components }}
+      aria-errormessage={error && 'folder-parent-error'}
```
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/components/ReactSelect/ReactSelect.js:20]: The error message doesn't make sense here: `ReactSelect` is meant to be a very generic component, which doesn't have anything todo with folders ...

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js [hunk 5]
```diff
@@ -129,18 +127,18 @@ export const EditFolderDialog = ({ onClose, folder, parentFolderId }) => {
 
   return (
     <>
-      <ModalLayout onClose={() => onClose()} labelledBy="title">
-        <EditFolderModalHeader isEditing={isEditing} />
+      <ModalLayout onClose={() => !isEditFolderLoading && onClose()} labelledBy="title">
```
Comment (gu-stav) [inline packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:130]: Why shouldn't I be able to close the modal, if the request is still running?

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js [hunk 6]
```diff
@@ -229,54 +222,54 @@ export const EditFolderDialog = ({ onClose, folder, parentFolderId }) => {
...
-                  {formatMessage({
-                    id: 'modal.folder.create.delete',
-                    defaultMessage: 'Delete folder',
-                  })}
-                </Button>
-              )}
+              <ModalFooter
+                startActions={
+                  <Button
+                    onClick={() => onClose()}
```
Comment (gu-stav) [inline packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:230]: In case `!isEditFolderLoading && onClose()` makes sense (which I don't think) it would need to be added here too.

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js [hunk 7]
```diff
@@ -229,54 +222,54 @@ export const EditFolderDialog = ({ onClose, folder, parentFolderId }) => {
...
+                }
+                endActions={
+                  <Stack horizontal spacing={2}>
+                    {isEditing && canUpdate && (
+                      <Button
+                        type="button"
+                        variant="danger-light"
+                        onClick={() => setShowConfirmDialog(true)}
+                        name="delete"
+                        disabled={isEditFolderLoading}
```
Comment (gu-stav) [inline packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:246]: To disable the button is not really a good a11y practice, as the element doesn't receive any events anymore. Shouldn't this use `loading={isEditFolderDeleteLoading}`?

File: packages/core/upload/admin/src/components/SelectTree/SelectTree.js [hunk 8]
```diff
@@ -45,6 +51,7 @@ const SelectTree = ({ options: defaultOptions, maxDisplayDepth, defaultValue, ..
       components={{ Option }}
       options={options}
       defaultValue={defaultValue}
+      error={error}
```
Comment (gu-stav) [inline packages/core/upload/admin/src/components/SelectTree/SelectTree.js:54]: Why can't it be spread through `props`?

File: packages/core/upload/admin/src/hooks/__mocks__/useEditFolder.js [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export const useEditFolder = jest.fn().mockReturnValue({
+  editFolder: jest.fn().mockResolvedValue({}),
```
Comment (gu-stav) [inline packages/core/upload/admin/src/hooks/__mocks__/useEditFolder.js:2]: Since this is a very general mock now, could you please return a realistic folder object instead, so that we test a real-world scenario here?

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 5 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:130]: because the request won't be cancel when closing, you can try it - it will close the modal then a few moments later will show the success. I can leave it like it was before but I thought it was cleaner this way

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 6 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:230]: Actually the user can't access the onClick event because the Save button is in loading state, and the Cancel / Delete buttons are on disabled state

image

File: packages/core/upload/admin/src/hooks/__mocks__/useEditFolder.js (hunk 9 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/hooks/__mocks__/useEditFolder.js:2]: sure - I would prefer to add tests in a future PR though, I would like to prioritize the ML folder fix we have to do before the release, WDYT?

Comment (gu-stav) [reply packages/core/upload/admin/src/hooks/__mocks__/useEditFolder.js:2]: Fine with me!

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 7 above)
Comment (gu-stav) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:247]: I agree about not having 3 spinners. What do think about that:

- close: not disabled, no loading state, because clicking it does not involve any risk, because no request is being fired
- delete / submit: loading state on the button

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 5 above)
Comment (gu-stav) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:130]: Tbh. I don't think it is a problem. I'd put it the other way around: if the request takes a while it is confusing for the user if nothing is happening, because there is no visual feedback, so they might actually click several times ...

File: packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js (hunk 7 above)
Comment (ronronscelestes) [reply packages/core/upload/admin/src/components/EditFolderDialog/EditFolderDialog.js:247]: works for me!

**Review by gu-stav — APPROVED** [review]
